### PR TITLE
Add custom default base address option to crossgen

### PR DESCRIPTION
**Pull request by gbalykov**
This PR allows to manually setup default base address option for native images during `crossgen`

```
crossgen /BaseAddress <base_addr_value> <other_options>
```

Also, env variable could be setup to enable/disable mapping of images at default base address.

cc @alpencolt @jkotas

### CODE REVIEW

File: src/pal/src/init/pal.cpp [hunk 1]
```diff
@@ -382,6 +385,20 @@ Initialize(
         }
 #endif // ENSURE_PRIMARY_STACK_SIZE
 
+        char* useDefaultBaseAddr = getenv("COMPlus_UseDefaultBaseAddr");
```
Comment (jkotas) [inline src/pal/src/init/pal.cpp]: This switch disables a OS security feature. We cannot have this config switch in the product.

Comment (gbalykov) [reply src/pal/src/init/pal.cpp]: Could this be added under some `#ifdef`?

Comment (jkotas) [reply src/pal/src/init/pal.cpp]: Putting it under ifdef like `FEATURE_ENABLE_NO_ADDRESS_SPACE_RANDOMIZATION` should be ok (similar to existing `FEATURE_ENABLE_NO_RANGE_CHECKS`).

**Comment by jkotas** [comment]
Could you please share some details on how you expect to use it? You have done a lot of work on converting pointers to relative pointers earlier. I would expect that it has equivalent benefit.

**Comment by gbalykov** [comment]
We want to enable both optimizations on removal of relocations and support for default base address option in the same build of CoreCLR. This will allow to choose different approaches to removal of relocations in different scenarios.

File: src/pal/src/init/pal.cpp (hunk 1 above)
Comment (gbalykov) [reply src/pal/src/init/pal.cpp]: I've put setup and usage of this parameter under `ifdef FEATURE_ENABLE_NO_ADDRESS_SPACE_RANDOMIZATION`

File: src/tools/crossgen/crossgen.cpp [hunk 2]
```diff
@@ -155,6 +155,7 @@ void PrintUsageHelper()
        W("                           input assemblies\n")
 
 #endif
+       W("    /BaseAddress <value> - Specifies base address to use for compilation.\n")
```
Comment (jkotas) [inline src/tools/crossgen/crossgen.cpp]: Could you please put this under the ifdef as well?

File: src/tools/crossgen/crossgen.cpp [hunk 3]
```diff
@@ -544,6 +546,17 @@ int _cdecl wmain(int argc, __in_ecount(argc) WCHAR **argv)
             fLargeVersionBubbleSwitch = true;
         }
 #endif
+        else if (MatchParameter(*argv, W("BaseAddress")))
```
Comment (jkotas) [inline src/tools/crossgen/crossgen.cpp]: And this place (the rest of the plumbing can stay)

Comment (gbalykov) [reply src/tools/crossgen/crossgen.cpp]: Done

**Comment by gbalykov** [comment]
Thanks!

**Pull request merged**